data_steward/retraction/retract_utils.py

```diff
@@ -79,6 +84,65 @@ def get_src_id(mapping_type):
     return src_id
 
 
+def get_datasets_list(project_id, dataset_ids):
+    # initialize list of all datasets in project
+    all_dataset_ids = [
+        dataset.dataset_id for dataset in bq.list_datasets(project_id)
+    ]
+
+    if not dataset_ids or dataset_ids == consts.NONE:
+        dataset_ids = []
+    elif dataset_ids == consts.ALL_DATASETS:
+        dataset_ids = all_dataset_ids
+    else:
+        dataset_ids = dataset_ids.split()
+        # only consider datasets that exist in the project
+        dataset_ids = [
+            dataset_id for dataset_id in dataset_ids
+            if dataset_id in all_dataset_ids
+        ]
+
+    # filter out datasets not of interest
+    dataset_ids = [
+        dataset_id for dataset_id in dataset_ids
+        if get_dataset_type(dataset_id) != common.OTHER and
+        not is_sandbox_dataset(dataset_id) and
+        not is_staging_dataset(dataset_id)
+    ]
+
+    logging.info('Found datasets to retract from: %s' % ', '.join(dataset_ids))
```

Comment:
I think we are preferring f-strings to formatting strings.  Suggest using:

```suggestion
    logging.info(f"Found datasets to retract from: {', '.join(dataset_ids)}")
```